Feature/todo list items

shubham14p3 opened the pull request:
In this project, we are building TODO List with javascript vanilla, ES6 and webpack.

## Project milestones

- The app ‘todos’ is going to be objects that will allow users to dynamically create, which means either using factories or constructors/classes to generate them.
- The Todo list should have a title, description, dueDate and priority. You might also want to include notes or even a checklist.
- Todo list should have projects or separate lists of todos. When a user first opens the app, users should be some sort of ‘default’ project to which all of their todos are put. 
- Users should be able to create new projects and choose which project their todos go into.
- Use localStorage to save user’s projects and todos between sessions.

## Project Specifics:

# User Interaction UI
- User can view all projects.
- Users can view all todos in each project.
- Users can expand a single todo to see/edit its details.
- Users can delete a todo.
- Users can toggle between complete and uncomplete by just clicking the check box.
- Users can add a project title under which they want to add the todo list.

## General requirements:
- [x] Add a descriptive README file to your repositories as described here.
- [x] Use meaningful names for your Pull Requests and add a short summary of its content as described here.
- [x] Use descriptive commit messages as described here.
- [x] Do not commit files that are not related to the project. Instead, add them to .gitignore file.
- [x] Do not commit inline comments. They will make your project look messy. If you need to review a previous version of your code, you can always use git history.

## Git Flow:
- [x] Use a specific branch for each feature.
- [x] Use a descriptive name for your branches.
- [x] Use Stylelint and ESlLint to validate your code.

shubham14p3 committed:
add linter branch

Cyrus-Kiprop committed:
initiate npm and configure github actions

Cyrus-Kiprop committed:
:pushpin: Linter files

add stylint, eslint and stickler config files

shubham14p3 committed:
add webpack and webpack branch

shubham14p3 committed:
fixed linter

shubham14p3 committed:
add html structure to index.html file

shubham14p3 committed:
added html structure to index.html file

Cyrus-Kiprop committed:
:lipstick: reset css

add css reset

Cyrus-Kiprop committed:
:wrench: Webpack loaders

add css, sass, scss, and file loaders

Cyrus-Kiprop committed:
configure webpack for development build

Cyrus-Kiprop committed:
add helper function to create HTML node elements

Cyrus-Kiprop committed:
:sparkles: Header

add an header to the UI

shubham14p3 committed:
add footer component

shubham14p3 committed:
add footer links

shubham14p3 committed:
add styling to footer

shubham14p3 committed:
add more footer styling

Cyrus-Kiprop committed:
:art: Abstract functionality

avoid repetition by abstracting creation and appending of childrens to parent elements

Cyrus-Kiprop committed:
:lipstick: header

add styling to the header

Cyrus-Kiprop committed:
:sparkles: Aside section

add aside section and styling

Cyrus-Kiprop committed:
:sparkles: Aside

finalize on the aside section and added some styles

shubham14p3 committed:
add todolist content

shubham14p3 committed:
add todo item default content

Cyrus-Kiprop committed:
:bug: Index.html file reset

fix index.html file reset by webpack

Cyrus-Kiprop committed:
:card_file_box: Data

add more todoList items to the mock database

Cyrus-Kiprop committed:
:recycle: Refactor todoList section

abstract data pasting function into pasteData method

Cyrus-Kiprop committed:
:sparkles: Filter

add filter by project basis functionality

Cyrus-Kiprop committed:
:lipstick: UI styling

update the todoListItem and aside styling

shubham14p3 committed:
add project title

shubham14p3 committed:
add title function to hearder

shubham14p3 committed:
add localStorage to project

Cyrus-Kiprop committed:
:sparkles: Add project functionality

add the add project functionality

Cyrus-Kiprop committed:
add delete list item functionality

Cyrus-Kiprop committed:
add delete project functionality

Cyrus-Kiprop committed:
:lipstick: TodoList Input

add forms/input

Cyrus-Kiprop committed:
finalize on add todo list item functionality

Cyrus-Kiprop committed:
add toggle functionality

Cyrus-Kiprop committed:
:ambulance: finalize on styling

add more styling to the UI

shubham14p3 committed:
removed console.log from entire project

shubham14p3 committed:
removed in line comments

shubham14p3 committed:
removed in line comments

shubham14p3 committed:
removed in line comments

shubham14p3 committed:
fixed stylelint in component

shubham14p3 committed:
fixed stylelint in assets

shubham14p3 committed:
add logo

Cyrus-Kiprop committed:
add validation to the inputs

Cyrus-Kiprop committed:
:ambulance: Fix toggle issues

fix toggle issues and add validation

shubham14p3 committed:
add default data

shubham14p3 committed:
Merge branch 'feature/todoListItems' of https://github.com/Cyrus-Kiprop/Todo-List into feature/todoListItems

shubham14p3 committed:
add data to correct format

Cyrus-Kiprop committed:
:bug: toggle issue

fix even toggling

shubham14p3 committed:
add images for readme

shubham14p3 committed:
updated readme file

Cyrus-Kiprop committed:
fix eslint and stylint errors

Cyrus-Kiprop committed:
Merge branch 'feature/todoListItems' of github.com:Cyrus-Kiprop/Todo-List into feature/todoListItems

Cyrus-Kiprop committed:
fix eslint and stylint errors

shubham14p3 committed:
add default data with right styling

Cyrus-Kiprop committed:
Initial dist subtree commit

shloch reviewed (changes requested):
## Hello @shubham14p3 , @Cyrus-Kiprop 

Hello guys, am very very sorry i can't test your app, I get this error below
I see screenshots in your README, means it works maybe

So maybe I have configurations/compatibility issues at my level, please re-check too at your level

If it's on your side, then fix, 
Also, please deploy online and put a live demo on the README, so that if it continues for another TSE, he might review via the live DEMO

![todo](https://user-images.githubusercontent.com/40712459/87363021-266a6900-c568-11ea-86bc-b68ec57a1e63.gif)


Thanks for understanding

Happy coding

Cyrus-Kiprop committed:
:ambulance: LocalStorage [DB]

initialize the localstorage with some data

shubham14p3 committed:
Merge branch 'feature/todoListItems' of https://github.com/Cyrus-Kiprop/Todo-List into feature/todoListItems

shubham14p3 committed:
add live links to read me files

bellom reviewed (approved):
## Hi Team :clap: 

**Great work making all the necessary corrections from the previous review.**

## Congratulation :+1:  
_Go ahead and merge to dev and don't forget to submit project completion form.
Good luck in the next project :slightly_smiling_face:_ 

**Best,**

**Oluwaseun Bello**
**_Slack Handle - @sean bellom_**

shubham14p3 commented:
Thanks for the Feedback,
All changes have been noted and implemented,
-We don't see this error so far in in our side, make sure you are on the correct branch thats is feature/tofolistitems
- Added live link to the readme file.
Note: If  you are still facing the issue with local repo, do let us know  with how to reproduce it so that we can analyze it further.
